great_expectations/metrics/domain.py

```diff
@@ -58,3 +70,26 @@ class ColumnValues(Values):
     """
 
     column: NonEmptyString
+
+    @classmethod
+    @override
+    def internal_name(cls):
+        return "column_map"
+
+
+class Batch(Domain):
```

Comment:
@billdirks What do you think about the `Batch` vs `Table` naming convention? Crazy idea...what if this class should be called `Table` and the class above it should be called `Batch` (the only attribute on that class is `batch_id`).

Reply:
Or what if all of these inherit from the existing `Batch` class!

Reply:
`Table` seems off to me, since this domain applies to lots of things which aren't tables. And even when it does apply to an actual table, row conditions and partitioners mean we're not actually validating the entire table. Seems like a more accurate description of the domain is "Whole Batch".